gitlabform/processors/project/branches_processor.py

```diff
@@ -136,3 +133,7 @@ def transform_branch_config(self, branch_config: dict):
                             item["group_id"] = self.gl.get_group_id(item.pop("group"))
 
         return branch_config
+
+    @staticmethod
+    def check_for_wildcard(branch):
+        return re.fullmatch(".*\\*.*", branch)
```

Comment:
I'm not too familiar with this. Could you help me understand how it helps to validate if the string contains wildcard or not? Is there a sample or explanation somewhere that I could read?

Reply:
This is just a simple regex match to check if there is a `*` in the branch name. 

I used the examples in the gitlab docs [here](https://docs.gitlab.com/ee/user/project/repository/branches/protected.html#protect-multiple-branches-with-wildcard-rules) to test this 

```
*-stable
production/*
*gitlab*
```

When trying to cut a branch in gitlab containing a `*` you get the error `Branch name cannot contain '*'`

Do you think I should add some comments around this?

Reply:
> When trying to cut a branch in gitlab containing a * you get the error Branch name cannot contain '*'

I believe that's a different scenario. A branch **name** cannot be `*`. But, you can created branch protection **rule** using `*`, which I believe is going to target every branch in the repo.

<img width="1197" alt="image" src="https://github.com/user-attachments/assets/e5cb1805-6837-4741-9212-b0aca08f7f48">

Reply:
> This is just a simple regex match to check if there is a * in the branch name.

If we just want to check for the presence of wildcard characters in the name, then we can do this: `return '*' in branch_name or '?' in branch_name`. Your logic currently doesn't check for the `?` character, which seems to be accepted by GitLab when creating a branch protection rule.

<img width="1116" alt="image" src="https://github.com/user-attachments/assets/faeb558b-c7b6-48f2-a650-ed9de00bb1fe">

Maybe we don't need to use regular expression? I think that might be helpful if we want to validate the wildcard pattern itself, but that logic should match how gitlab validates a wildcard pattern. Maybe we don't need to go that far? What do you think?

Reply:
I believe the code caters for this scenario as the branch name still contains a `*` so the regex returns true and still falls down the correct code path, does that make sense? I'm struggling to think of a valid test scenario

Reply:
In this scenario what does the '?' do? I can't find reference to this anywhere in the gitlab documentation. Its also not possible to create a branch with a '?' in the **name** so I'm unsure how this fits in.

I see your point about not having to use a regex match though. I've pushed a change for this. 

I think the code currently caters for a protected branch that is just called '*' but I can't think of a valid test scenario to add for this.

Reply:
> Do you have a link to any documentation about using '?'

https://www.geeksforgeeks.org/wildcard-pattern-matching/

Does that help?

Reply:
That does help, I think I understand how the '?' works in wildcard pattern matching but I can't get it to work inside gitlab at all. I definitely could be doing something wrong here. I have a repo here that has a currently open branch called `r-1-3-0` should the below protection not match that branch? 

![image](https://github.com/user-attachments/assets/ab414d55-3067-480a-9983-74f1b87fe996)

Reply:
I think `?-*` should match `r-1-3-0`. At least it does when I try it here...

https://www.digitalocean.com/community/tools/glob?comments=true&glob=%3F-%2A&matches=false&tests=r-1-3-0

But I'm not sure what gitlab does under the hood. Their doc shows simple example like you mentioned earlier and there's no reference to their wildcard pattern matching. At least I didn't find it. Is it possible that they don't actually support `?` in the pattern? 🤷 . Maybe we just don't add the validation of `?` in the test. Clearly GitLab allows creating protection rule using `?` and so we keep the logic to make sure gitlabform allows it.

Reply:
Yeah I'm really not sure to be honest, I've added the logic into the code to be able to cater for it and all the tests are still passing.

Reply:
Sounds good. Thanks for all the work and tests on this PR.